### PR TITLE
feat(table): refactor table to new styles

### PR DESCRIPTION
**Pull request by zvonimirfras**

### CODE REVIEW

File: src/table/table.component.ts [hunk 1]
```diff
@@ -328,12 +341,11 @@ import { getScrollbarWidth } from "../common/utils";
...
 	/**
 	 * Size of the table rows.
-	 * (size `"default"` is being deprecated as of neutrino v1.2.0, please use `"md"` instead)
 	 *
-	 * @type {("sm" | "md" |"default" | "lg")}
+	 * @type {("sm" | "md" | "lg")}
 	 * @memberof Table
 	 */
-	@Input() size: "sm" | "md" |"default" | "lg" = "md";
+	@Input() size: "sm" | "md" | "lg" = "md";
```
[the remark this answers is not in the document]
Comment (zvonimirfras) [reply src/table/table.component.ts]: It is, I have all 3 in stories.

File: src/table/table-model.class.ts [hunk 2]
```diff
@@ -285,11 +319,20 @@ export class TableModel {
 		const rri = this.realRowIndex(index);
 		this.data.splice(rri, 1);
 		this.rowsSelected.splice(rri, 1);
+		this.rowsExpanded.splice(rri, 1);
 		this.rowsContext.splice(rri, 1);
```
[the remark this answers is not in the document]
Comment (zvonimirfras) [reply src/table/table-model.class.ts]: This function is fairly short and the same code isn't repeated anywhere else so there's no real benefit from it now.

With the new new designs and feature set, we might need to refactor model as well to catch up with the features and then it's worth looking at.

File: src/table/table.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cal-smith) [reply src/table/table.component.ts]: Must have missed that, cool :+1:

File: src/table/table.stories.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+	sort(model, index: number) {
+		if (model.header[index].sorted) {
+			// if already sorted flip sorting direction
+			model.header[index].ascending = model.header[index].descending;
+		}
+		model.sort(index);
+	}
+
+	addRow() {
+		const lastRowCopy = JSON.parse(JSON.stringify(this.model.row(this.model.data.length - 1)));
```
Comment (theiliad) [inline src/table/table.stories.ts]: Should we move this into a Tools file?

This is what I'm doing in charts
image

File: src/table/table.stories.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+		this.model.data = [
+			[
+				new TableItem({data: "Name 1", expandedData: "No template"}),
+				new TableItem({data: {name: "Lessy", link: "#"}, template: this.customTableItemTemplate})
+			],
+			[new TableItem({
+				data: "Name 3",
+				expandedData: {name: "In", surname: "Template"},
+				expandedTemplate: this.customTableItemTemplate
+			}), new TableItem({data: "swer"})],
```
Comment (theiliad) [inline src/table/table.stories.ts]: New line after [ & before ] for all entries?

File: src/table/table.stories.ts [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+				[striped]="striped">
+			</app-custom-table>
+		`,
+		props: {
+			size: selectV2("size", {Small: "sm", Normal: "md", Large: "lg"}, "md", "table-size-selection"),
+			showSelectionColumn: boolean("showSelectionColumn", true),
+			striped: boolean("striped", true)
+		}
+	}));
+
```
Comment (theiliad) [inline src/table/table.stories.ts]: Can also do this to distinguish each story more:
image

File: src/checkbox/checkbox.component.ts [hunk 6]
```diff
@@ -56,25 +54,23 @@ export class CheckboxChange {
...
+			[required]="required"
+			[checked]="checked"
+			[disabled]="disabled"
+			[indeterminate]="indeterminate"
+			[attr.aria-label]="ariaLabel"
+			[attr.aria-labelledby]="ariaLabelledby"
+			[attr.aria-checked]="indeterminate ? 'mixed' : checked"
+			(change)="onChange($event)"
+			(click)="onClick($event)">
+		<label [for]="id" class="bx--checkbox-label"><ng-content></ng-content></label>
```
Comment (theiliad) [inline src/checkbox/checkbox.component.ts]:
```html
<label [for]="id" class="bx--checkbox-label">
   <ng-content></ng-content>
</label>
```
?

File: src/table/table.stories.ts (hunk 5 above)
Comment (zvonimirfras) [reply src/table/table.stories.ts]: Can, shouldn't in this case. This follows what other stories in this repo do.

We can consider refactoring all of them.

**Pull request merged**